plugins/modules/mso_rest.py

```diff
@@ -0,0 +1,245 @@
+- name: Add tenant
+  cisco.mso.mso_rest:
+    host: mso
+    username: admin
+    password: SomeSecretPassword
+    path: /api/v1/tenants
+    method: post
+    content:
+      {
+          "displayName": "mso_tenant",
+          "name": "mso_tenant",
+          "description": "",
+          "siteAssociations": [],
+          "userAssociations": [],
+          "_updateVersion": 0
+      }
+  delegate_to: localhost
+
+- name: Add schema (check_mode)
+  cisco.mso.mso_rest:
+    host: mso
+    username: admin
+    password: SomeSecretPassword
+    path: /api/v1/schemas
+    method: post
+    content:
+      {
+          "displayName": "mso_schema",
+          "templates": [{
+              "name": "Template_1",
+              "tenantId": "{{ add_tenant.jsondata.id }}",
+              "displayName": "Template_1",
+              "templateSubType": [],
+              "templateType": "stretched-template",
+              "anps": [],
+              "contracts": [],
+              "vrfs": [],
+              "bds": [],
+              "filters": [],
+              "externalEpgs": [],
+              "serviceGraphs": [],
+              "intersiteL3outs": []
+          }],
+          "sites": [],
+          "_updateVersion": 0
+      }
+  delegate_to: localhost
+
+- name: Query schema
+  cisco.mso.mso_rest:
+    host: mso
+    username: admin
+    password: SomeSecretPassword
+    path: /api/v1/schemas
+    method: get
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+'''
+
+import json
+import os
+
+try:
+    from ansible.module_utils.six.moves.urllib.parse import parse_qsl, urlencode, urlparse, urlunparse
+    HAS_URLPARSE = True
+except Exception:
+    HAS_URLPARSE = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec
+from ansible.module_utils.urls import fetch_url
+from ansible.module_utils._text import to_text
+
+
+def update_qsl(url, params):
+    ''' Add or update a URL query string '''
+
+    if HAS_URLPARSE:
+        url_parts = list(urlparse(url))
+        query = dict(parse_qsl(url_parts[4]))
+        query.update(params)
+        url_parts[4] = urlencode(query)
+        return urlunparse(url_parts)
+    elif '?' in url:
+        return '?' + urlencode(params)
+    else:
+        return url + '?' + '&'.join(['%s=%s' % (k, v) for k, v in params.items()])
+
+
+class MSORESTModule(MSOModule):
+
+    def changed(self, d):
+        ''' Check MSO response for changes '''
+
+        if isinstance(d, dict):
+            for k, v in d.items():
+                if k == 'status' and v in ('created', 'modified', 'deleted'):
+                    return True
+                elif self.changed(v) is True:
+                    return True
+        elif isinstance(d, list):
+            for i in d:
+                if self.changed(i) is True:
+                    return True
+
+        return False
+
+    def response_type(self, rawoutput, rest_type='json'):
```

Comment:
if this is just calling response_json and because MSO only has JSON API, should we skip this and directly use response_json?